sdk/storage/azdatalake/directory/client_test.go

```diff
@@ -61,6 +62,30 @@ type UnrecordedTestSuite struct {
 	suite.Suite
 }
 
+func (d *UnrecordedTestSuite) TestDirNewSubdirectoryClient() {
+	_require := require.New(d.T())
+	testName := d.T().Name()
+
+	accountName, _ := testcommon.GetGenericAccountInfo(testcommon.TestAccountDatalake)
+	_require.Greater(len(accountName), 0)
+
+	svcClient, err := testcommon.GetServiceClient(d.T(), testcommon.TestAccountDatalake, nil)
+	_require.NoError(err)
+
+	fsName := testcommon.GenerateFileSystemName(testName)
+	fsClient := svcClient.NewFileSystemClient(fsName)
+
+	dirName := testcommon.GenerateDirName(testName)
+	dirClient := fsClient.NewDirectoryClient(dirName)
+
+	subdirName := "subdir"
+	subdirClient, err := dirClient.NewSubdirectoryClient(subdirName)
+	_require.NoError(err)
+
+	correctDirURL := fmt.Sprintf("https://%s.dfs.core.windows.net/%s/%s/%s/", accountName, fsName, dirName, subdirName)
```

Comment:
Let's add like a create call and maybe a couple other calls and make this a Recorded test